src/database/database.go

```diff
@@ -0,0 +1,102 @@
+package database
+
+import (
+	"fmt"
+	"log"
+	_ "github.com/lib/pq" //PostgreSQL driver
+	"database/sql"
+	"github.com/garyburd/redigo/redis"
+)
+
+var (
+	RedisPool *redis.Pool // creating redis pool enables us to reuse redigo connections
+	SQL *sql.DB
+	databases Info
+)
+
+// Type is the type of database from a Type* constant
+type Type string
+
+const (
+	TypePostgreSQL Type = "PostgreSQL"
+	TypeRedis Type = "Redis"
+)
+
+type Info struct {
+	// Database type
+	Type[] Type
+	// Postgres info if used
+	PostgreSQL PostgreSQLInfo
+	Redis RedisInfo
+
+}
+
+// PostgreSQLInfo is the details for the database connection
+type PostgreSQLInfo struct {
+	Hostname  string
+	Port  int
+	DatabaseName string
+	Username string
+	Password string
+}
+type RedisInfo struct {
+	URL string
+	Port int
+}
+
+
+// DSN returns the Data Source Name
+func DSN(ci PostgreSQLInfo) string {
+	return fmt.Sprintf("host=%s port=%d user=%s "+
+		"password=%s dbname=%s sslmode=disable",
+		ci.Hostname, ci.Port, ci.Username, ci.Password, ci.DatabaseName)
+}
+func DSN_Redis(ci RedisInfo) string {
+	return fmt.Sprintf("%s"+":"+"%d",ci.URL,ci.Port)
+
+}
+
+
+// Connect to the database
+func Connect(d Info){
+	var err error
+
+	// Store the config
+	databases = d
+
+	for index,_:= range d.Type[:]{
+		switch d.Type[index] {
+		case TypePostgreSQL:
```

Comment:
It is better to make two types DB and Cache and its methods Connect instead of switch